wolfcrypt/src/port/iotsafe/iotsafe.c

```diff
@@ -1172,13 +1300,55 @@ int wc_iotsafe_ecc_verify_hash(byte *sig, word32 siglen, byte *hash,
     ret = wc_ecc_sig_to_rs(sig, siglen, sig_raw, &r_size, sig_raw
             + IOTSAFE_ECC_KSIZE, &s_size);
     if (ret == 0)
-        ret = iotsafe_verify_hash(key_id, IOTSAFE_HASH_SHA256, IOTSAFE_SIGN_ECDSA,
+        ret = iotsafe_verify_hash(key_id, id_size, IOTSAFE_HASH_SHA256, IOTSAFE_SIGN_ECDSA,
             hash, hashlen, sig_raw, 2 * IOTSAFE_ECC_KSIZE, res);
     return ret;
 }
+
+int wc_iotsafe_ecc_gen_k_ex(byte *key_id, uint16_t id_size)
+{
+    int ret = 0;
+    ecc_key* key = XMALLOC(sizeof(ecc_key), NULL, DYNAMIC_TYPE_ECC);
+    ret = iotsafe_gen_keypair(key_id, id_size, key);
+    if (ret > 0)
+        ret = 0;
+    XFREE(key, NULL, DYNAMIC_TYPE_ECC);
+    return ret;
+}
+
+int wc_iotsafe_ecc_import_public(ecc_key *key, byte key_id)
+{
+    return iotsafe_get_public_key(&key_id, 1, key);
+}
+
+int wc_iotsafe_ecc_export_public(ecc_key *key, byte key_id)
+{
+    return iotsafe_put_public_key(&key_id, 1, key);
+}
+
+int wc_iotsafe_ecc_sign_hash(byte *in, word32 inlen, byte *out,
+        word32 *outlen, byte key_id)
+{
+    return iotsafe_sign_hash(&key_id, 1, IOTSAFE_HASH_SHA256, IOTSAFE_SIGN_ECDSA,
+            in, inlen, out, outlen);
+}
+
+
+int wc_iotsafe_ecc_verify_hash(byte *sig, word32 siglen, byte *hash,
+        word32 hashlen, int *res, byte key_id)
+{
+    return wc_iotsafe_ecc_verify_hash_ex(sig, siglen, hash, hashlen, res,
+            &key_id, 1);
+}
 int wc_iotsafe_ecc_gen_k(byte key_id)
 {
-    return iotsafe_gen_keypair(key_id);
+    int ret = 0;
+    ecc_key* key = XMALLOC(sizeof(ecc_key), NULL, DYNAMIC_TYPE_ECC);
+    ret = iotsafe_gen_keypair(&key_id, 1, key);
```

Comment:
Same here... Seems odd allocating an `ecc_key` struct to not be used and not initializing it.